lib/spack/spack/cmd/ci.py

```diff
@@ -676,14 +676,24 @@ def ci_rebuild(args):
     # outside of the pipeline environment.
     if install_exit_code == 0:
         if buildcache_mirror_url or pipeline_mirror_url:
-            spack_ci.create_buildcache(
-                env=env,
+            results = spack_ci.create_buildcache(
+                input_spec=job_spec,
                 buildcache_mirror_url=buildcache_mirror_url,
                 pipeline_mirror_url=pipeline_mirror_url,
                 pr_pipeline=spack_is_pr_pipeline,
-                json_path=job_spec_json_path,
             )
 
+            for result in results:
+                success = result["success"]
+                url = result["url"]
+                message = "{0} buildcache for {1}/{2} to {3}".format(
+                    "Successfully pushed" if success else "Failed to push",
```

Comment:
What exactly do you expect here? The only exception that ever causes this to be `False` is `NoOverwriteException` which happens when the spec.json or tarball already exist.